### ISSUE TITLE
Write a specification for taxonomy parser / renderer

### ISSUE THREAD

**Issue by alexgarel** [openfoodfacts/taxonomy-editor#7]
### What

- Specify how the .txt file should be parsed by the taxonomy editor to get target structure.

### Part of
- #5

**Comment by alexgarel** [comment]
@aadarsh-ram I wrote the specification (sorry for being that late, but taxonomy is more complex that I though)

For some part I put python code, because it's just a translation of perl code, and I think it was more exact that my formulation in plain english, so it avoids misinterpretations.

I think @stephanegigandet should review the PR before you consider it ok. @stephanegigandet feel free to commit any problem.

As I go through perl parser, I realized, it was not easy to just parse and build a json (and perl code is very complex because of that). That's why I propose you instead to parse and build the database on the fly, so that you can, later on, use the database to do the processing, like finding parents. It will be more memory efficient, even if maybe a bit slower (we shall see).

Good luck :-)

**Comment by aadarsh-ram** [comment]
Thank you for the documentation! I'll be reading through it extensively and will clarify any doubts that I might have.
